src/ctapipe/monitoring/aggregator.py

```diff
@@ -1,159 +1,390 @@
-is a monitoring table with columns describing the start and stop time of the chunk
-and the aggregated statistic values.
+These classes take as input an events table containing any event-wise quantities
+(e.g., images, scalars, vectors), divide it into time chunks, which may optionally
+overlap, and compute various aggregated statistics for each chunk. The statistics
+include the count, mean, median, and standard deviation. The result is a monitoring
+table with columns describing the start and stop time of the chunk and the
+aggregated statistic values.
+
+The aggregation is always performed along axis=0 (the event dimension), making
+these classes suitable for any N-dimensional event-wise data.
 """
 
 __all__ = [
+    "BaseAggregator",
     "StatisticsAggregator",
     "PlainAggregator",
     "SigmaClippingAggregator",
 ]
 
-from abc import abstractmethod
+from abc import ABC, abstractmethod
 from collections import defaultdict
 
+import astropy.units as u
 import numpy as np
 from astropy.stats import sigma_clip
 from astropy.table import Table
 
 from ctapipe.containers import StatisticsContainer
-from ctapipe.core import TelescopeComponent
-from ctapipe.core.traits import Int
+from ctapipe.core import Component
+from ctapipe.core.traits import CaselessStrEnum, Int
 
-__all__ = [
-    "StatisticsAggregator",
-    "PlainAggregator",
-    "SigmaClippingAggregator",
-]
 
-
-class StatisticsAggregator(TelescopeComponent):
+class BaseAggregator(Component, ABC):
     """
-    Base component to handle the computation of aggregated statistic values from a table
-    containing e.g. charges, peak times and/or charge variances (images).
+    Base class for aggregators that compute statistics over chunks of data.
+
+    Aggregators always return a Table with time windows and computed statistics.
+    Subclasses can add multiple columns by overriding add_result_columns().
     """
 
     chunk_size = Int(
-        2500,
-        help="Size of the chunk used for the computation of aggregated statistic values",
+        default_value=None,
+        allow_none=True,
+        help=(
+            "Size of the chunk used for the computation of aggregated statistic values. "
+            "If None, use the entire table as one chunk. "
+            "For event-based chunking: number of events per chunk. "
+            "For time-based chunking: duration in seconds per chunk (integer)."
+        ),
+    ).tag(config=True)
+
+    chunking_mode = CaselessStrEnum(
+        ["events", "time"],
+        default_value="events",
+        help=(
+            "Chunking strategy: 'events' for fixed event count, 'time' for time intervals. "
+            "When 'time', chunk_size and chunk_shift are interpreted as seconds."
+        ),
     ).tag(config=True)
 
     def __call__(
         self,
         table,
-        masked_pixels_of_sample=None,
+        masked_elements_of_sample=None,
         chunk_shift=None,
         col_name="image",
```

Comment:
see below: I'd add `time_tolerance = 0.1 * u.s` to customise based on the use case.